insta-clone

LDinho opened the pull request.

LDinho committed:
initial commit

LDinho committed:
add initial comment, search & post components set up

LDinho committed:
update dummy-data file; set up access to data

LDinho committed:
add sass and prop-types packages; initial sass set up

LDinho committed:
update sass files; add font awesome icons

LDinho committed:
add PostContainer component with passing data; update styling & dummy-data file

LDinho committed:
add CommentSection and Comment components with styling

LDinho committed:
add SearchBar component and styling

LDinho committed:
add comment input element to CommentSection component

LDinho committed:
add componentDidMount to App.js; implement add new comments functionality for each post

LDinho committed:
refactor PostContainer to create a Post component

LDinho committed:
add liking post increment functionality to heart icon

LDinho committed:
add username search functionality to display its posts

LDinho committed:
fix bug- prevent showing empty state if no search match found

LDinho committed:
pass data from children components up to App component

WIP- This is because comments are being reset during the search process

LDinho committed:
pass likes data up to App to prevent likes reset on clearing search

theryanclausen commented on instagram/src/components/SearchBar/SearchBar.js:
```diff
@@ -0,0 +1,67 @@
+  }
+
+  handleInputChange = (event) => {
+    console.log('EVENT', event.target.value)
```
Look at all of this extra logic you are using to pass the searchbar value back up to app. Is it necessary to store searchQuery here in local state?

theryanclausen commented on instagram/src/components/SearchBar/SearchBar.js:
```diff
@@ -0,0 +1,67 @@
+    if (searchQuery) return;
+
+    // When there's no search value or the clear button is pressed in the input field
+    this.props.handleSubmit(searchQuery);
```
Another thing to keep in mind is that you can have an input without a form. Your searchQuery string can be held in state without a submit. That's what the changehandler is doing

LDinho committed:
refactor search- make SearchBar component uncontrolled; update icon tags to self-closing

LDinho committed:
refactor- move code from App.js to PostPage.js; add a fixed SearchBar/nav to top of page

PostPage.js now renders in App.js.

LDinho committed:
update search filtering of username to use include()

search filters and starts matching as a user types the search query

LDinho committed:
set up initial HOC; change search input width

LDinho committed:
add create account & login/authentication functionality using localStorage

- current loggedIn user's username now displays when current user adds comments.
- create account message pops up when user does not exist

LDinho committed:
add log out functionality

theryanclausen commented on instagram/src/components/LoginPage/LoginPage.js:
```diff
@@ -0,0 +1,92 @@
+          <label htmlFor="password">Password</label>
+          <input className="password"
+                 id="password"
+                 type="text"
```
try using input type password

theryanclausen commented on instagram/src/components/LoginPage/LoginPage.js:
```diff
@@ -0,0 +1,92 @@
+  };
+
+  handleInputChange = (event) => {
+    const {value, name} = event.target;
```
nice job

theryanclausen commented on instagram/src/App.js:
```diff
@@ -0,0 +1,27 @@
+const ComponentFromWithAuthenticate = withAuthenticate(PostPage)(LoginPage);
+
+class App extends Component {
+  constructor(props) {
```
does this component need a constructor?

theryanclausen commented on instagram/src/components/PostContainer/PostPage.js:
```diff
@@ -0,0 +1,97 @@
+  }
+
+  logOut = () => {
+    this.props.onLogout();
```
you don't need to do this . It's OK to pass your onLogout straight from props without creating a method on this component

theryanclausen reviewed (commented):
nice job. Just be careful with reassigning that logout method to different names so many times. It is easy to lose track of it that way. I would stick with logout or onLogout. There are use cases for reassigning methods to different names, but that mostly involves reusable components.

LDinho committed:
refactor- remove unneeded code

LDinho committed:
[refactor styling] add styled-components to SearchBar

LDinho committed:
[refactor style] add styled-components to PostContainer

LDinho committed:
MVP- add PropTypes
